internal/util/dataplane/mode.go

```diff
@@ -1,12 +1,32 @@
 package dataplane
 
+type DBMode string
+
+const (
+	DBModeOff       = "off"
+	DBModePostgres  = "postgres"
+	DBModeCassandra = "cassandra"
+)
+
 // IsDBLessMode can be used to detect the proxy mode (db or dbless).
-func IsDBLessMode(mode string) bool {
-	return mode == "" || mode == "off"
+func IsDBLessMode(mode DBMode) bool {
+	return mode == "" || mode == DBModeOff
 }
 
 // DBBacked returns true if the gateway is DB backed.
 // reverse of IsDBLessMode for readability.
-func DBBacked(mode string) bool {
+func DBBacked(mode DBMode) bool {
 	return !IsDBLessMode(mode)
 }
```

Comment:
These functions could be methods of `DBMode` for convenience or we could simply drop them if we only have two valid values of the enum (`DBModeOff` and `DBModeOn`) that we could use with equality operator like:

```go
if dbMode == dataplane.DBModeOff { ... }
```

I'd lean towards the latter.

Reply:
Sometimes dbmode is left empty. It would be hard to search for them all, if we left over any of them, it may cause a bug.

Reply:
My point is that if we had the `DBMode` enum defined as `type DBMode struct { mode string }`, it wouldn't be possible, and `==` could be used safely.